application/helpers/kalkun_helper.php

```diff
@@ -381,3 +380,59 @@ function is_null_loose($input)
 	}
 	return empty($input);
 }
+
+/**
+ * Convert a phone number as input by the user to E164 format
+ * using the region of the user.
+ * Done with libphonenumber
+ *
+ * @param string $phone
+ * @return string
+ */
+function phone_format_e164($phone)
+{
+	$CI = &get_instance();
+	// reformat phone number to E164
+	$phoneNumberUtil = \libphonenumber\PhoneNumberUtil::getInstance();
+	$region = $CI->Kalkun_model->get_setting()->row('country_code');
+	$phoneNumberObject = $phoneNumberUtil->parse($phone, $region);
```

Comment:
What happens if this parse fails? Given that on the UI there is a bypass feature, that would fail here.

Reply:
User will have the exception dispayed on screen (as a CI3 error) with the corresponding error:, something like "NOT a number" or any other exception message returned by the lib.
I think we should not hide it.

Reply:
In fact no, not on these screens. It will just block (js popup to compose, add user... stays on screen) Unfortunately kalkun doesn't have any support for when there is an error in sending a message (whether db insertion error or whatever). IIRC  I saw such a bug report in the past.

Reply:
With #313, the user can't enter number considered as invalid by this exact same library since the check is done by the same lib in both case (the php on that is on the server)